pkg/proc/variables.go

```diff
@@ -1063,59 +1060,61 @@ func (v *Variable) structMember(memberName string) (*Variable, error) {
 			v = &v.Children[0]
 		}
 	}
-	structVar := v.maybeDereference()
-	structVar.Name = v.Name
-	if structVar.Unreadable != nil {
-		return structVar, nil
-	}
 
-	switch t := structVar.RealType.(type) {
-	case *godwarf.StructType:
-		for _, field := range t.Field {
-			if field.Name != memberName {
-				continue
-			}
-			return structVar.toField(field)
-		}
-		// Check for embedded field only if field was
-		// not a regular struct member
-		for _, field := range t.Field {
-			isEmbeddedStructMember :=
-				field.Embedded ||
-					(field.Type.Common().Name == field.Name) ||
-					(len(field.Name) > 1 &&
-						field.Name[0] == '*' &&
-						field.Type.Common().Name[1:] == field.Name[1:])
-			if !isEmbeddedStructMember {
-				continue
-			}
-			// Check for embedded field referenced by type name
-			parts := strings.Split(field.Name, ".")
-			if len(parts) > 1 && parts[1] == memberName {
+	queue := []*Variable{v}
+	seen := map[string]struct{}{} // prevent infinite loops
+	first := true
+
+	for len(queue) > 0 {
```

Comment:
This changeset seems more like a refactor than dead code elimination. Would you mind sending this patch in separate?

Reply:
Ah, it seems you have. I'll take a look at the other PR first then, perhaps we can rebase this after.

Reply:
Yep. I think I messed up a rebase, actually. Fixed it now.